js/client.js

```diff
@@ -2475,10 +2477,54 @@
 			this.close();
 			app.focusRoom('');
 			app.rooms[''].focusPM(this.data.name);
+		},
+		toggleOptions: function () {
+			app.addPopup(UserOptionsPopup, {name: this.data.name, userid: this.data.userid});
 		}
 	}, {
 		dataCache: {}
 	});
+	
+	var UserOptionsPopup = this.UserOptions = Popup.extend({
+		initialize: function (data) {
+			this.name = data.name.substr(1);
+			this.userid = data.userid;
+			this.update();
+		},
+		update: function () {
+			var buf = '<p><button name="toggleIgnoreUser">' + (app.ignore[this.userid] ? 'Unignore' : 'Ignore') + '</button></p>';
+			buf += '<p><button name="rank">Rank</button></p>';
+			this.$el.html(buf);
+		},
+		toggleIgnoreUser: function () {
+			var buf = "User '" + this.name + "'";
+			if (app.ignore[this.userid]) {
+				delete app.ignore[this.userid];
+				buf += " no longer ignored.";
+			} else {
+				app.ignore[this.userid] = 1;
+				buf += " ignored. (Moderator messages will not be ignored.)";
+			}
+			var $pm = $('.pm-window-' + this.userid);
+			if ($pm.length && $pm.css('display') !== 'none') {
+				$pm.find('.inner').append('<div class="chat">' + buf + '</div>');
+			} else {
+				var room = (app.curRoom && app.curRoom.add ? app.curRoom : app.curSideRoom);
+				if (!room || !room.add) {
+					app.addPopupMessage(buf);
+					return this.update();
+				}
+				room.add(buf);
+			}
+			app.dismissPopups();
+		},
+		rank: function () {
+			var room = (app.curRoom && app.curRoom.parseCommand ? app.curRoom : app.curSideRoom);
+			if (!room || !room.parseCommand) return app.addPopupMessage("You must join a room to use this command.");
```

Comment:
See, this is just ugly, clicking on the username doesn't have weird gotchas like this.